File: maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {

        int max_so_far = INT_MIN;
        int max_sum = 0;

        for(int i=0;i<nums.size();i++){
            max_sum += nums[i];
            if(max_so_far<max_sum)
                max_so_far = max_sum;
            if(max_sum < 0)
            max_sum =0;
        }

        return max_so_far;
        
    }
};

File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {

        //Kadane's algorithm

        int max_so_far =  Integer.MIN_VALUE;
        int curr_sum = 0;

        for(int i=0;i<nums.length;i++){
            curr_sum+= nums[i];
            max_so_far = Math.max(max_so_far,curr_sum);
            if(curr_sum<0)
            curr_sum=0;
        }

        return max_so_far;
        
    }
}